docs/fraud-proofs.md

```diff
@@ -0,0 +1,63 @@
+
+## Rollup header
+
+Rollups can adopt many approaches to prove that fraud happened. One of which could be having the following fields in the rollup header:
+
+- Rollup block state root
+- A sequence of spans in Celestia: which references where the rollup data was published in the Celestia chain.
+
+Note: The sequence of spans can be defined using the following: `Height`, `start index`, and `length` in the Celestia block, in the case of a single Celestia block. However, it could be generalized to span over multiple blocks.
+
+For the rest of the document, we will suppose that the sequence of spans only references one Celestia block.
+
+## Unavailable data fraud proof
+
+By construction, the rollup block data is the sequence of spans defined in the header. Thus, proving that the data is unavailable goes back to proving that the sequence of spans doesn't belong to the Celestia block, i.e. the span is out of bounds.
+
+We could prove that via creating a binary [merkle proof](https://github.com/celestiaorg/celestia-core/blob/c3ab251659f6fe0f36d10e0dbd14c29a78a85352/crypto/merkle/proof.go#L19-L31) of any row/column to the Celestia data root. This proof will provide the `Total` which is the number of rows/columns in the extended data square. And, we can use that to calculate the square size.
+
+Then, we will use that information to check if the provided transaction index, in the header, is out of the square size bounds.
+
+For the data root, we will use a binary merkle proof to prove its inclusion in a data root tuple root that was committed to by the QGB smart contract. More on this in [here](#1-data-root-inclusion-proof).
+
+## Invalid transaction fraud proof
+
+In order to prove an invalid transaction in the rollup, we need to prove the following:
+
+- Prove that that transaction was posted to Celestia
+- Prove that the transaction is invalid: left to the rollup to define.
+
+The first part, proving that the transaction was posted to Celestia, can be done in three steps:
+
+1. Prove that the data root is committed to by the QGB smart contract
+2. Inclusion proof of the transaction to Celestia data root
+3. Prove that that transaction is in the rollup sequence spans
+
+### 1. Data root inclusion proof
+
+To prove the data root is committed to by the QGB smart contract, we will need to provide a merkle proof of the data root tuple to a data root tuple root. This can be created using the [`data_root_inclusion_proof`](https://github.com/celestiaorg/celestia-core/blob/c3ab251659f6fe0f36d10e0dbd14c29a78a85352/rpc/client/http/http.go#L492-L511) query.
+
+### 2. Transaction inclusion proof
+
+To prove that a rollup transaction is part of the data root, we will need to provide two proofs: a namespace merkle proof of the transaction to a row root. This could be done via proving the shares that contain the transaction to the row root using a namespace merkle proof. And, a binary merkle proof of the row root to the data root.
```

Comment:
```suggestion
To prove that a rollup transaction is part of the data root, we will need to provide two proofs: a namespace Merkle proof of the transaction to a row root. This could be done via proving the shares that contain the transaction to the row root using a namespace Merkle proof. And, a binary Merkle proof of the row root to the data root.
```